File: 3768-check-if-digits-are-equal-in-string-after-operations-i/check-if-digits-are-equal-in-string-after-operations-i.java
class Solution {
    public boolean hasSameDigits(String s) {
        while (s.length() > 2) {
            StringBuilder newS = new StringBuilder();

            for (int i = 0; i < s.length() - 1; i++) 
                newS.append((s.charAt(i) - '0' + s.charAt(i + 1) - '0') % 10);

            s = newS.toString();
        }

        return s.charAt(0) == s.charAt(1);
    }
}

File: 3768-check-if-digits-are-equal-in-string-after-operations-i/check-if-digits-are-equal-in-string-after-operations-i.go
func hasSameDigits(s string) bool {
	currList := []byte(s)
	insertPos := 0
    
	for len(currList) > 2 {
		insertPos = 0
		for i := 1; i < len(currList); i++ {
			currList[insertPos] = (byte(int(currList[i-1]-'0')+int(currList[i]-'0')) % 10) + '0'
			insertPos++
		}
		currList = currList[:len(currList)-1]
	}

	return currList[0] == currList[1]
}